[move reference] Review new docs (#35)
MystenLabs/move-book: reference/src/packages.md

address_to_be_filled_in = "0x101010101"
```

Most of the sections in the package manifest are self explanatory, but named addresses can be a bit
difficult to understand so we examine them in more detail in
[Named Addresses During Compilation](#named-addresses-during-compilation), but before that we'll

> first take a look at the `Move.lock` file and what it contains.

Reviewer: IMO this is a confusion transition. I would consider swapping the two sections (since the lockfile doesn't appear to be mentioned in the address discussion) and rewrite this as:

> Most of the sections in the package manifest should be self explanatory, with the exception of named addresses which are explained in the 
[next section](#named-addresses-during-compilation). After being built, the `Move.toml` and configuration settings are stored in the [`Move.lock` file](#move.lock).

Author: I'll give it a shot!

Reviewer 2: I like this a lot more than the current layout, so 👍🏻 from me on moving this section around.

flavor = <string> # The flavor of the Move compiler used to build the package, e.g. "sui"
```

With that, let's now turn to the compilation process and how named addresses are resolved, and how

> to use them.

Reviewer: Will need to change this if we do the reordering

Additionally, every named address in a package is exported. Because of this and the above scoping
rules each package can be viewed as coming with a set of named addresses that will be brought into

> scope when the package is imported, e.g., if the `example_pkg` package was imported, that
> importation would bring into scope the `named_addr` named address. Because of this, if `P` imports

Reviewer: ```suggestion
scope when the package is imported, e.g., if you import `example_pkg`, that
import will also bring the `named_addr` named address into scope. Because of this, if `P` imports
```

## Usage and Artifacts

The Move package system comes with a command line option as part of the CLI:
`sui move <command> <command_flags>`. Unless a particular path is provided, all package commands
will run in the current working directory. The full list of commands and flags for the Move CLI can

> be found by running `sui move --help`.

Reviewer: This seems a little unclear, because, e.g., `sui move test` will work at any directory in a package.

Reviewer 2: Yea this is a great callout -- this should be something like (or similar)
```
... Unless a particular path is provided, all package commands will run in the current enclosing Move package. ....
```

### Artifacts

A package can be compiled using CLI commands. This will create a `build` directory containing

> build-related artifacts (such as bytecode, source maps, and documentation). The general layout of

Reviewer: ```suggestion
build-related artifacts (including bytecode, source maps, and documentation). The general layout of
```
